src/compiler/types.ts

```diff
@@ -6785,6 +6785,8 @@ namespace ts {
 
     /* @internal */
     export interface ParenthesizerRules {
+        getParenthesizeLeftSideOfBinaryForOperator(binaryOperator: SyntaxKind): (leftSide: Expression) => Expression;
+        getParenthesizeRightSideOfBinaryForOperator(binaryOperator: SyntaxKind): (rightSide: Expression) => Expression;

```

Comment:
These are new, and exist to support passing in a `(node: Node) => Node` parenthesizer rule for binary expressions. The implementation simply wraps `parenthesizeLeftSideOfBinary` or `parenthesizeRightSideOfBinary` and caches the result for reuse.